File: Programing_Basics_with_JS/forLoopMoreExercises/04.Grades.js
function grades(input) {
    let index = 0;
    let numbersOfStudents = Number(input[index]);
    index++;

    let failStudents = 0;
    let between3And4Students = 0;
    let between4And5Students = 0;
    let topStudents = 0;
    let allScore = 0;


    for (let i = 1; i <= numbersOfStudents; i++) {
        let studentScore = Number(input[index]);
        index++;

        if (studentScore < 3) {
            failStudents++;
            allScore += studentScore;
        } else if (studentScore < 4) {
            between3And4Students++;
            allScore += studentScore;
        } else if (studentScore < 5) {
            between4And5Students++;
            allScore += studentScore;
        } else {
            topStudents++;
            allScore += studentScore;
        }
    }

    let averageScore = allScore / numbersOfStudents;
    let failStudentsPercent = failStudents / numbersOfStudents * 100;
    let between3And4StudentsPercent = between3And4Students / numbersOfStudents * 100;
    let between4And5StudentsPercent = between4And5Students / numbersOfStudents * 100;
    let topStudentsPercent = topStudents / numbersOfStudents * 100;

    console.log(`Top students: ${topStudentsPercent.toFixed(2)}%`);
    console.log(`Between 4.00 and 4.99: ${between4And5StudentsPercent.toFixed(2)}%`);
    console.log(`Between 3.00 and 3.99: ${between3And4StudentsPercent.toFixed(2)}%`);
    console.log(`Fail: ${failStudentsPercent.toFixed(2)}%`);
    console.log(`Average: ${averageScore.toFixed(2)}`);
}

grades(['10' , '3.00', '2.99', '5.68', '3.01', '4', '4', '6', '4.50', '2.44', '5'])
grades(['6' , '2', '3', '4', '5', '6', '2.2'])

File: Programming_Basics_with_Python/for_loop_more_exercises/04._grades.py
students = int(input())

total_score = 0
top_students = 0
students_between_four_and_five = 0
students_between_three_and_four = 0
failed_students = 0

for s in range(1, students + 1):
    student_score = float(input())

    total_score += student_score

    if student_score >= 5:
        top_students += 1
    elif student_score >= 4:
        students_between_four_and_five += 1
    elif student_score >= 3:
        students_between_three_and_four += 1
    else:
        failed_students += 1

print(f'Top students: {top_students / students * 100:.2f}%')
print(f'Between 4.00 and 4.99: {students_between_four_and_five / students * 100:.2f}%')
print(f'Between 3.00 and 3.99: {students_between_three_and_four / students * 100:.2f}%')
print(f'Fail: {failed_students / students * 100:.2f}%')
print(f'Average: {total_score / students:.2f}')